### PR TITLE
Add assume role policy conditions and managed iam policies

### LINKED ISSUE #24: Allow for condition on role assumption
## Describe the Feature
Add the variable `condition`, what this does is if present it will append the role assumption policy with a condition. This allows for fine tuning what can assume a role.

## Expected Behavior
Ability to provide conditions to an assume role policy.

## Use Case
The most simple use case I can think of for this is role assumption via an external IDP like Okta. You are assuming the role via a SAML process.

The policy should look like this:
```
{
  "Version": "2012-10-17",
  "Statement": [
    {
      "Sid": "",
      "Effect": "Allow",
      "Principal": {
        "Federated": "arn:aws:iam::1234567890123:saml-provider/Okta"
      },
      "Action": "sts:AssumeRoleWithSAML",
      "Condition": {
        "StringEquals": {
          "SAML:aud": "https://signin.aws.amazon.com/saml"
        }
      }
    }
  ]
}
```

## Describe Ideal Solution
Accept a `condition` block and if present then use it in the data block for generating the aws iam json policy.

## Alternatives Considered
The only alternative at this time after looking through the code for this module is to use the basic terraform aws_iam_role for this.

### PR DESCRIPTION
**Pull request by sebastianmacarescu**
## what
* option to attach AWS Managed IAM policies to created role
* option to add conditions to trust policy

## why
* we should be able to use aws managed policies (or any other policies) and not create new ones dedicated for this role
* we should be able to add conditions on who can assume this role (mfa enabled, be part of organization, specific session name, etc)

## references
* closes #24
* documentation for conditions: https://registry.terraform.io/providers/hashicorp/aws/latest/docs/data-sources/iam_policy_document
* documentation for conditions in trust role policies: https://aws.amazon.com/blogs/security/how-to-use-trust-policies-with-iam-roles/

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -47,6 +56,13 @@ resource "aws_iam_role_policy_attachment" "default" {
   policy_arn = join("", aws_iam_policy.default.*.arn)
 }
 
+resource "aws_iam_role_policy_attachment" "managed" {
+  for_each   = module.this.enabled ? var.managed_policy_arns : []
```
Comment (nitrocode) [inline main.tf:60]: Will this work without a set() around the list ?

Comment (sebastianmacarescu) [reply main.tf:60]: for each works only with maps or sets

Comment (sebastianmacarescu) [reply main.tf:60]: i dind't test, we can change to set()

Comment (sebastianmacarescu) [reply main.tf:60]: @nitrocode I have tested this and it works without calling toset()

**Review by nitrocode — APPROVED** [review]

**Pull request merged**